lib/validate.js

```diff
@@ -0,0 +1,90 @@
+var allowedParams = {
+	"common": ["type", "products", "currency"],
+	"impression": ["list"],
+	"click": ["list"],
+	"detail": ["list"],
+	"add": [],
+	"remove": [],
+	"purchase": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"refund": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"checkout": ["step", "option"],
+	"checkout_option": ["step", "option"]
+};
+
+var allowedProductParams = {
+	"common": ["id", "name", "brand", "category", "variant", "price"],
+	"impression": ["position"],
+	"click": ["position", "coupon"],
+	"detail": ["position", "coupon"],
+	"add": ["quantity", "coupon"],
+	"remove": ["quantity", "coupon"],
+	"purchase": ["quantity", "coupon"],
+	"refund": ["quantity", "coupon"],
+	"checkout": ["quantity", "coupon"],
+	"checkout_option": ["quantity", "coupon"]
+};
+
+function validateHit(ecData, cb) {
```

Comment:
I understand that `cb` is  common  parameter name in *JS* world, but I'd preferred something like `handleError[Cb]`.